packages/opal-common/opal_common/git/bundle_maker.py

```diff
@@ -53,19 +53,61 @@ def _get_explicit_manifest(self, viewer: CommitViewer) -> Optional[List[str]]:
         loaded into OPA.
 
         This method searches for an explicit manifest file, reads it and returns the list of paths, or None if not found.
-        """
-        def is_manifest_file(f: VersionedFile) -> bool:
-            return f.path == Path(self._manifest_filename)
 
-        manifest_files = list(viewer.files(is_manifest_file))
-        if not manifest_files:
-            return None
+        The manifest file can include references to other directories containing a ".manifest" file,
+        those would be recursively expanded to compile the final manifest list.
+        """
+        visited_paths = []
+
+        def _compile_manifest_file(dir: VersionedDirectory, manifest_file_name: str = ".manifest") -> List[str]:
+            explicit_manifest = list()
+            try:
+                manifest_file = viewer.get_file(dir.path / manifest_file_name)
+                assert manifest_file is not None, "Manifest file not found"
+
+                for path_entry in manifest_file.read().splitlines():
+                    # Path is relative to current directory, make it absolute
+                    path_entry = dir.path / path_entry
+                    if not viewer.exists(path_entry):
+                        logger.warning(f"Path '{path_entry}' does not exist") 
```

Comment:
These log entries will become hard to understand/debug if we're dealing with a large, complex set of deeply-nested manifests. It would be better if it included the full path of how we got here, from the root manifest.